pkg/kubelet/status/status_manager.go

```diff
@@ -481,6 +490,12 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 	glog.V(3).Infof("Status for pod %q updated successfully: (%d, %+v)", format.Pod(pod), status.version, status.status)
 	m.apiStatusVersions[kubetypes.MirrorPodUID(pod.UID)] = status.version
 
+	// Clean up exited containers
+	err = m.podTryCleanupResources.PodTryCleanupResources(pod, status.status)
```

Comment:
this moves logic that feels like it should happen in `syncPod` or `syncLoopIteration` into status manager which blurs responsibility across boundaries in the kubelet.  i would prefer we find an alternative approach to the problem.  did you evaluate other options that integrated with the sync loop path?

Reply:
Yes.  At least for what I tried, it wound up taking longer than the approach I chose, because it was still waiting for the sync loop interval (up to 10 seconds).  That's better than the 1 minute of the GC interval, but still results in considerably more delay that cleaning the pods up synchronously with with delete request (which averages 2-3 seconds).